examples/mnist/advance_model_training.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Multiple Model Training Example
+# 
+# This example trains multiple models and extracts training statistics
+
+# ## Import required libraries
+from ludwig.api import LudwigModel
+from ludwig.visualize import learning_curves
+import logging
+import shutil
+import os
+import glob
+import yaml
+from collections import namedtuple
+
+# clean out old results
+try:
+    shutil.rmtree('./results')
+except:
+    pass
+
+try:
+    shutil.rmtree('./visualizations')
+except:
+    pass
+
+try:
+    file_list = glob.glob('./data/*.json')
+    file_list.append(glob.glob('./data/*.hdf5'))
+    os.remove(file_list)
+except:
+    pass
+
+# read in base model definition
+with open('./model_definition.yaml', 'r') as f:
+    base_model = yaml.safe_load(f.read())
+
+# Specify named tuple to keep track of training results
+TrainingResult = namedtuple('TrainingResult', ['name', 'train_stats'])
```

Comment:
Could you please change the names of variables (everywhere) to be lowercased and use _ as separators? like `training_results`.

Reply:
I understand you'd like me to change `TrainingResult` to be `training_result`.  

Before making this change, I'd like to explain the rationale for `TrainingResult`.  If after reading this rationale you still want the change, I'll make the change.

`TrainingResult` is defined by the factory function `collections.namedtuple()`.   I viewed this a new class definition so followed a the class naming convention in PEP8.  This is similar to the example code found in the [`namedtuple()` api documentation](https://docs.python.org/3/library/collections.html#collections.namedtuple), e.g.,
```
EmployeeRecord = namedtuple('EmployeeRecord', 'name, age, title, department, paygrade')
```
This is why I used CamelCase for `TrainingResult`. 

Do you still want me to change `TrainingResult` to 'training_result`?

Reply:
I see, so that's a factory of constructors. Understood, let's keep it like it is.